pkg/action/diff.go

```diff
@@ -164,55 +183,101 @@ func fileDestination(ctx context.Context, c bincapz.Config, fr, tr *bincapz.File
 	}
 }
 
-type diffReports struct {
-	Added     string
-	AddedFR   *bincapz.FileReport
-	Removed   string
-	RemovedFR *bincapz.FileReport
+// filterMap filters orderedmap pairs by checking for matches against a slice of compiled regular expression patterns.
+func filterMap(om *orderedmap.OrderedMap[string, *bincapz.FileReport], ps []*regexp.Regexp, c chan<- *orderedmap.Pair[string, *bincapz.FileReport], g *errgroup.Group) {
+	for pair := om.Oldest(); pair != nil; pair = pair.Next() {
+		pair := pair
+		g.Go(func() error {
+			for _, pattern := range ps {
+				if match := pattern.FindString(filepath.Base(pair.Key)); match != "" {
+					c <- pair
+				}
+			}
+			return nil
+		})
+	}
 }
 
-// combineReports builds a flattened slice of added and removed paths and their respective file reports.
-func combineReports(d *bincapz.DiffReport) []diffReports {
-	diffs := make(chan diffReports)
-	var wg sync.WaitGroup
-
-	for removed := d.Removed.Oldest(); removed != nil; removed = removed.Next() {
-		wg.Add(1)
-		go func(path string, fr *bincapz.FileReport) {
-			defer wg.Done()
-			for added := d.Added.Oldest(); added != nil; added = added.Next() {
-				diffs <- diffReports{
-					Added:     added.Key,
-					AddedFR:   added.Value,
-					Removed:   path,
-					RemovedFR: fr,
+// combine iterates over the removed and added channels to create a diff report to store in the combined channel.
+func combine(removed, added <-chan *orderedmap.Pair[string, *bincapz.FileReport], combined chan<- bincapz.CombinedReport, g *errgroup.Group) {
+	for r := range removed {
+		for a := range added {
+			a := a
+			r := r
+			g.Go(func() error {
+				score := levenshtein.Match(r.Key, a.Key, levenshtein.NewParams())
+				if score < 0.9 {
+					return nil
 				}
-			}
-		}(removed.Key, removed.Value)
+				combined <- bincapz.CombinedReport{
+					Added:     a.Key,
+					AddedFR:   a.Value,
+					Removed:   r.Key,
+					RemovedFR: r.Value,
+					Score:     score,
+				}
+				return nil
+			})
+		}
+	}
+}
+
+// combineReports orchestrates the population of the diffs channel with relevant diffReports.
+func combineReports(c bincapz.Config, d *bincapz.DiffReport, combined chan<- bincapz.CombinedReport) error {
+	defer close(combined)
+
+	maxConcurrency := c.Concurrency
+	if maxConcurrency < 1 {
+		maxConcurrency = 1
 	}
 
-	go func() {
-		wg.Wait()
-		close(diffs)
-	}()
+	var g errgroup.Group
+	g.SetLimit(maxConcurrency)
+
+	// Patterns we care about when handling diffs
+	patterns := []string{
+		`^[\w.-]+\.(so|a|dylib|dll|lib)$`,
```

Comment:
I believe we only need `.so` here - the other extension styles don't typically changed based on the version.

Reply:
Updated in [`e9fb45e` (#429)](https://github.com/chainguard-dev/bincapz/pull/429/commits/e9fb45e8d59d0604d032540b37708bbe892b5df2#diff-296f8d8ff6837443670f31f7e7dd79d8908c01dc9059790f2ab6d1c78cafacdbR232).